hoss/coordinate_utilities.py

```diff
@@ -0,0 +1,232 @@
+""" This module contains utility functions used for
+    coordinate variables and methods to convert the
+    coordinate variable data to x/y dimension scales
+"""
+
+from typing import Set, Tuple
+
+import numpy as np
+from netCDF4 import Dataset
+from numpy import ndarray
+from varinfo import VariableFromDmr, VarInfoFromDmr
+
+from hoss.exceptions import (
+    CannotComputeDimensionResolution,
+    IrregularCoordinateVariables,
+    MissingCoordinateVariable,
+)
+
+
+def get_override_projected_dimension_names(
+    varinfo: VarInfoFromDmr, variable_name: str
+) -> str:
+    """returns the x-y projection variable names that would
+    match the group of geo coordinate names. The  coordinate
+    variable name gets converted to 'projected_y' dimension scale
+    and 'projected_x'
+
+    """
+    override_variable = varinfo.get_variable(variable_name)
+
+    if override_variable is not None and (
+        override_variable.is_latitude() or override_variable.is_longitude()
+    ):
+        projected_dimension_names = [
+            f'{override_variable.group_path}/projected_y',
+            f'{override_variable.group_path}/projected_x',
```

Comment:
I guess I could just have group_path/coordinate_name_projected_x  or projected_y
The coordinate names would be unique
it would be latitude_projected_x or latitude_projected_y . It would not matter . it would be unique

Reply:
One approach (that would make this function more complicated 😞) could be to:

* (Already done) - Get the `VariableFromDmr` instance.
* Get the references from the `coordinates` metadata attribute.
* Find the latitude and longitude variables from those references and put them in a list. - Maybe reusing your `get_coordinate_variables` function.
* Make a unique hash of those references.
* Prepend that hash on to `.../projected_x` and `.../projected_y` (like you're already doing with the group path).

I took some inspiration from [MaskFill](https://git.earthdata.nasa.gov/projects/SITC/repos/maskfill/browse/pymods/MaskFillUtil.py#240). To try something like:

```
from hashlib import sha256

# Faking the coordinates list for now:
coordinates_list = ['/latitude', '/longitude']

coordinates_hash = sha256(' '.join(coordinates_list).encode('utf-8')).hexdigest()

projected_dimension_names = [
    f'{coordinates_hash}/projected_y',
    f'{coordinates_hash}/projected_x',
]
```

The string will look a bit ugly (e.g., 'a089b9ebff6935f6c8332710de2ee3b351bd47c1fb807b22765969617027e8d2'), but it will be unique and reproducible.